### PR TITLE
Allow instrumenting OpenAI/Anthropic client classes or modules

### PR DESCRIPTION
**Pull request by alexmojaki**
For a user who wanted to just `logfire.instrument_openai()` without passing a client, since they want to instrument langchain which doesn't expose/accept an openai client object: https://pydanticlogfire.slack.com/archives/C06EDRBSAH3/p1715977569378699

### CODE REVIEW

**Comment by alexmojaki** [comment]
@willbakst might be interested in reviewing too.

**Review by willbakst — APPROVED** [review]
LGTM

File: logfire/_internal/integrations/llm_providers/llm_provider.py [hunk 1]
```diff
@@ -22,15 +23,55 @@ def instrument_llm_provider(
     scope_suffix: str,
     get_endpoint_config_fn: Callable[[Any], EndpointConfig],
     on_response_fn: Callable[[Any, LogfireSpan], Any],
-    is_async_client_fn: Callable[[Any], bool],
+    is_async_client_fn: Callable[[type[Any]], bool],
 ) -> ContextManager[None]:
-    """Instruments the provided `client` with `logfire`."""
+    """Instruments the provided `client` (or clients) with `logfire`.
+
+    `client` can be:
```
Comment (Kludex) [inline logfire/_internal/integrations/llm_providers/llm_provider.py]:
```suggestion
    The `client` can be:
```

File: logfire/_internal/integrations/llm_providers/llm_provider.py [hunk 2]
```diff
@@ -22,15 +23,55 @@ def instrument_llm_provider(
...
+    - a single client instance, e.g. an instance of `openai.OpenAI`.
+    - a class of a client
+    - an iterable of clients/classes.
```
Comment (Kludex) [inline logfire/_internal/integrations/llm_providers/llm_provider.py]:
```suggestion
    - a single client instance, e.g. an instance of `openai.OpenAI`;
    - a class of a client;
    - an iterable of clients/classes.
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**